qiskit_nature/problems/second_quantization/lattice/lattice/lattice.py

```diff
@@ -0,0 +1,354 @@
+                an edge payload and return a string label for the edge. For example::
+
+                    edge_labels=str
+
+                could be used to just return a string cast of the edge's data payload.
+                Or something like::
+
+                    edge_labels=lambda edge: edge['label']
+
+                could be used if the edge payloads are dictionaries. If this is set
+                edge labels will be drawn in the visualization.
+            font_size: An optional font size to use for text labels, By
+                default a value of 12 is used for nodes and 10 for edges.
+            font_color: An optional font color for strings. By default
+                ``'k'`` (i.e. black) is set.
+            font_weight: An optional string used to specify the font weight.
+                By default a value of ``'normal'`` is used.
+            font_family: An optional font family to use for strings. By
+                default ``'sans-serif'`` is used.
+            label: An optional string label to use for the graph legend.
+            connectionstyle: An optional value used to create a curved arc
+                of rounding radius rad. For example,
+                ``connectionstyle='arc3,rad=0.2'``. See
+                :class:`matplotlib.patches.ConnectionStyle` and
+                :class:`matplotlib.patches.FancyArrowPatch` for more info. By default
+                this is set to ``"arc3"``.
+            self_loop: Draw self-loops in a lattice. Defaults to False.
+            **kwargs: Kwargs for drawing the lattice.
+
+        Returns:
+            A matplotlib figure for the visualization if not running with an
+            interactive backend (like in jupyter) or if ``ax`` is not set.
+        """
+        if self_loop:
+            mpl_draw(
+                graph=self.graph,
+                pos=pos,
+                ax=ax,
+                arrows=arrows,
+                arrowstyle=arrowstyle,
+                arrow_size=arrow_size,
+                with_labels=with_labels,
+                node_list=node_list,
+                edge_list=edge_list,
+                node_size=node_size,
+                node_color=node_color,
+                node_shape=node_shape,
+                alpha=alpha,
+                cmap=cmap,
+                vmin=vmin,
+                vmax=vmax,
+                linewidths=linewidths,
+                width=width,
+                edge_color=edge_color,
+                edge_cmap=edge_cmap,
+                edge_vmin=edge_vmin,
+                edge_vmax=edge_vmax,
+                style=style,
+                labels=labels,
+                edge_labels=edge_labels,
+                font_size=font_size,
+                font_color=font_color,
+                font_weight=font_weight,
+                font_family=font_family,
+                label=label,
+                connectionstyle=connectionstyle,
+                **kwargs,
+            )
```

Comment:
This entire block is already present in the super class and is duplicated another time right below. I would extract it into smth like `HyberCubic._mpl(**kwargs)` and pass all parameters via a dict. Any of the `draw()` methods can then simply gather their named inarguments in a dict and pass it in.

In general, maybe we could replace the duplication of all the named arguments in favor of a single `kwargs` input? I don't see a good reason for exposing them here.